Source/DafnyCore/AST/Members/MethodOrFunction.cs

```diff
@@ -39,10 +43,10 @@ public abstract class MethodOrFunction : MemberDecl {
   }
 
   public void ResolveMethodOrFunction(INewOrOldResolver resolver) {
-    
-    var hasPrecondition = Req.Count > 0 || this.Formals.Any(f => f.Type.AsSubsetType is not null);
-    if (hasPrecondition) {
-      
+    if (HasPrecondition && !Bodyless && this.IsExtern(resolver.Options)) {
```

Comment:
I think you also need to check that it doesn't have the `{:compile false}` attribute. Sometimes Dafny functions are given a body, as a way to show that their specs are consistent (i.e., can be realized by at least one possible implementation), but then treated as imported during compilation.

Ultimately, the distinction between imports and exports is kind of fuzzy right now, and it's something I hope we can clarify when we revamp the FFI.